internal/service/certificatesigningrequest.go

```diff
@@ -264,19 +261,33 @@ func (h *ServiceHandler) ApproveCertificateSigningRequest(ctx context.Context, r
 		if errors.Is(err, flterrors.ErrResourceNotFound) {
 			return server.ApproveCertificateSigningRequest404JSONResponse{}, nil
 		}
-		return nil, err
+		return server.ApproveCertificateSigningRequest500JSONResponse{Message: err.Error()}, nil
+	}
+	if csr == nil {
+		return server.ApproveCertificateSigningRequest500JSONResponse{Message: "CertificateSigningRequest is nil"}, nil
 	}
 
+	// do not approve a denied request, or recreate a cert for an already-approved request
 	if api.IsStatusConditionTrue(csr.Status.Conditions, api.CertificateSigningRequestDenied) {
 		return server.ApproveCertificateSigningRequest409JSONResponse{Message: "The request has already been denied"}, nil
 	}
 	if api.IsStatusConditionTrue(csr.Status.Conditions, api.CertificateSigningRequestApproved) {
 		return server.ApproveCertificateSigningRequest409JSONResponse{Message: "The request has already been approved"}, nil
 	}
 
-	signedCert, err := signApprovedCertificateSigningRequest(h.ca, csr)
+	signedCert, err := signApprovedCertificateSigningRequest(h.ca, *csr)
 	if err != nil {
-		return nil, err
+		switch {
+		case errors.Is(err, flterrors.ErrCNLength) ||
+			errors.Is(err, flterrors.ErrInvalidPEMBlock) ||
+			errors.Is(err, flterrors.ErrSignature) ||
+			errors.Is(err, flterrors.ErrCSRParse) ||
+			errors.Is(err, flterrors.ErrSignCert) ||
+			errors.Is(err, flterrors.ErrEncodeCert):
```

Comment:
I think the golang style is to use commas:

```suggestion
		case errors.Is(err, flterrors.ErrCNLength), errors.Is(err, flterrors.ErrInvalidPEMBlock),  errors.Is(err, flterrors.ErrSignature), errors.Is(err, flterrors.ErrCSRParse), errors.Is(err, flterrors.ErrSignCert), errors.Is(err, flterrors.ErrEncodeCert):
```

Reply:
I am not sure this form is relevant here, since comma separated case values are for values - not for conditions

Reply:
So should I leave this as-is?